lib/Data/Validate/Sanctions.pm

```diff
@@ -32,22 +41,24 @@ sub is_sanctioned {
 }
 
 my @__data;
+
 sub __load_data {
-    return @__data if @__data;
+    my $stat = stat($sanction_file) or croak "Can't get stat of file $sanction_file, please check it.\n";
+    return @__data if ($stat->mtime <= $last_time && @__data);
 
-    my $file = __FILE__;
-    $file =~ s/\.pm/\.csv/;
-    open(my $fh, '<', $file) or croak "Can't find $file, please re-install the module.\n";
+    open(my $fh, '<', $sanction_file) or croak "Can't open file $sanction_file, please check it.\n";
     @__data = <$fh>;
     close($fh);
-
     chomp(@__data);
+    $last_time = $stat->mtime;
     @__data;
 }
 
 # for OO
 sub new {
     my $class = shift;
+    my %args  = @_;
+    $sanction_file = $args{sanction_file} if exists $args{sanction_file};
```

Comment:
If we're going to allow passing `sanction_file` to the constructor, it needs to be an instance variable, otherwise cases like this aren't going to work as expected:

```
my @checks = Data::Validate::Sanctions->new(sanction_file => $_), qw(list1.csv list2.csv);
for(@checks) {
  die "sanctioned user" if $_->is_sanctioned($first, $last);
}
```